### PR TITLE
Fixed type error when compiling for arm

### PR DESCRIPTION
**Pull request by PlutoTramble**
**This is the error that I got when I tried to compile for arm :**

InitialError

**Step to reproduce :**
This command inside src/raspberrypi directory:
`make clean && make EXTRA_FLAGS="-DSPDLOG_FMT_EXTERNAL -DFMT_HEADER_ONLY"`

### CODE REVIEW

File: src/raspberrypi/devices/scsi_powerview.cpp [hunk 1]
```diff
@@ -401,7 +401,7 @@ bool SCSIPowerView::Init(const map<string, string>& params)
 	fblinelen = fbfixinfo.line_length;
 
 	fb = (char *)mmap(0, fbfixinfo.smem_len, PROT_READ | PROT_WRITE, MAP_SHARED, fbfd, 0);
-	if ((int)fb == -1){
+	if (atoi(fb) == -1){
```
Comment (akuker) [inline src/raspberrypi/devices/scsi_powerview.cpp:404]: This doesn't look right. atoi is used to convert ascii strings to integers. mmap doesn't return a string.

How about:
`if(fb == (char*)-1){`

**Comment by akuker** [comment]
Thank you for the pull request @PlutoTramble !! Just one minor requested update.

> **Step to reproduce :** This command inside src/raspberrypi directory: `make clean && make EXTRA_FLAGS="-DSPDLOG_FMT_EXTERNAL -DFMT_HEADER_ONLY"`

Which OS were you running on when you got these errors? 64 bit Raspberry Pi OS? which version?

Thank you again for the pull request!!

File: src/raspberrypi/devices/scsi_powerview.cpp (hunk 1 above)
Comment (PlutoTramble) [reply src/raspberrypi/devices/scsi_powerview.cpp]: ` if(fb == (char*)-1){`
This seems to work!

**Comment by PlutoTramble** [comment]
No problem! Thank you for responding @akuker !

Yeah I should have been more exhaustive, my bad.

At first I had this error when I compiled on my raspberry pi 4 with Raspbian 11 (Bullseye) 64 bit. 
I was able to replicate it on my Linux desktop with Fedora 37 64 bit with the argument EXTRA_FLAGS="-DSPDLOG_FMT_EXTERNAL -DFMT_HEADER_ONLY".

I have to admit that I didn't test that one out for the Raspbian 32 bit version. Testing that right away.

**Comment by akuker** [comment]
> At first I had this error when I compiled on my raspberry pi 4 with Raspbian 11 (Bullseye) 64 bit. I was able to replicate it on my Linux desktop with Fedora 37 64 bit with the argument EXTRA_FLAGS="-DSPDLOG_FMT_EXTERNAL -DFMT_HEADER_ONLY".

Makes sense! I haven't really used 64bit OS's for Pi's yet. 

> I have to admit that I didn't test that one out for the Raspbian 32 bit version. Testing that right away.
No problem. This branch is so far out of date... its going to be a pain to merge it back into the mainline. (which is why I've been putting it off) I started working it in the `feature_powerview_merge`, but then got distracted by shiny things.

Thank you again for the PR. Feel free to submit more :)

**Pull request merged**